olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/TileRequest.h

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <boost/optional.hpp>
+#include <sstream>
+#include <string>
+
+#include <olp/core/geo/tiling/TileKey.h>
+#include <olp/dataservice/read/FetchOptions.h>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+/**
+ * @brief Encapsulates the fields required to request tile for the given
+ * key.
+ *
+ * You should specify tile key. Additionaly offset and depth could be set. If
+ * not set used default values, for offset default value is 2, for depth - 4.
+ */
+class DATASERVICE_READ_API TileRequest final {
+ public:
+  /**
+   * @brief Gets the billing tag to group billing records together.
+   *
+   * The billing tag is an optional free-form tag that is used for grouping
+   * billing records together. If supplied, it must be 4–16 characters
+   * long and contain only alphanumeric ASCII characters [A-Za-z0-9].
+   *
+   * @return The `BillingTag` string or `boost::none` if the billing tag is not
+   * set.
+   */
+  inline const boost::optional<std::string>& GetBillingTag() const {
+    return billing_tag_;
+  }
+
+  /**
+   * @brief Sets the billing tag for the request.
+   *
+   * @see `GetBillingTag()` for information on usage and format.
+   *
+   * @param tag The `BillingTag` string or `boost::none`.
+   *
+   * @return A reference to the updated `TileRequest` instance.
+   */
+  inline TileRequest& WithBillingTag(const boost::optional<std::string>& tag) {
```

Comment:
```suggestion
  inline TileRequest& WithBillingTag(const boost::optional<std::string> tag) {
```

and move internally.